src/sage/graphs/generic_graph.py

```diff
@@ -21014,6 +21016,95 @@ def layout_graphviz(self, dim=2, prog='dot', **options):
         positions = dot2tex.dot2tex(self.graphviz_string(**options), format='positions', prog=prog)
 
         return {key_to_vertex[key]: pos for key, pos in positions.items()}
+    
+    def layout_tutte(self, external_face, external_face_pos=None, **options):
+        """
+
+        Compute graph layout based on a Tutte embedding.
+
+        The graph must be 3-connected and planar.
+
+        INPUT:
+
+        - ``external_face`` -- list; the external face to be made a polygon
+
+        - ``external_face_pos`` -- list (default: ``None``); the positions of the vertices 
+          of the external face. If ``None``, will automatically generate a unit sided regular polygon.
+        
+        - ``**options`` -- other parameters not used here
+
+        OUTPUT: a dictionary mapping vertices to positions
+
+        EXAMPLES::
+
+            sage: g = graphs.WheelGraph(n=7)
+            sage: g.layout_tutte(external_face=[1, 2, 3, 4, 5, 6])
+            {1: (-0.499999999999999, 0.866025403784438),
+            2: (0.500000000000000, 0.866025403784439),
+            3: (1.00000000000000, 8.85026869717109e-17),
+            4: (0.500000000000000, -0.866025403784439),
+            5: (-0.500000000000001, -0.866025403784438),
+            6: (-1.00000000000000, 4.55896726715916e-16),
+            0: (-5.55111512312578e-17, 1.86944233679563e-16)}
+            sage: g = graphs.CubeGraph(n=3, embedding=2)
+            sage: g.plot(layout='tutte', external_face=['101','111','001','011'], external_face_pos={'101':(1,0), '111':(0,0), '001':(2,1), '011':(-1,1)})
+            Launched png viewer for Graphics object consisting of 21 graphics primitives
+
+        """
+        from sage.graphs.graph import Graph
+        from sage.matrix.constructor import zero_matrix
+        from sage.rings.real_mpfr import RR
+        
+        if (len(external_face) < 3):
+            raise ValueError("External face must have at least 3 vertices")
+
+        if (not self.is_planar()):
+            raise ValueError("Graph must be planar")
+
+        C = self.subgraph(vertices=external_face)
+        if (not C.is_cycle(directed_cycle=False)):
+            raise ValueError("External face must be a cycle")
+        external_face_ordered = C.depth_first_search(start=external_face[0], ignore_direction=False)
+        
+        from sage.graphs.connectivity import vertex_connectivity
+        if (not vertex_connectivity(self, k=3)):
+            raise ValueError("Graph must be 3-connected")
+
+        from math import sin, cos, pi
+        pos = dict()
+
+        if external_face_pos is None:
+            l = len(external_face)
+            a0 = pi/l+pi/2
```

Comment:
pycodestyle: `a0 = pi/l + pi/2`

also, it might be better to avoid variable `l` that is often confused with `1`.